src/material/core/option/option.scss

```diff
@@ -32,8 +32,19 @@
   &.mdc-list-item--disabled {
     // This is the same as `mdc-list-mixins.list-disabled-opacity` which
     // we can't use directly, because it comes with some selectors.
-    opacity: mdc-list-variables.$content-disabled-opacity;
     cursor: default;
+
+    // Reduce opacity of the contents of a disabled item to give a "grayed out" appearance. Avoid
+    // changing opacity of the option element itself to support focusing on disabled options using
```

Comment:
I don't quite follow the reasoning why we aren't greying out the entire option. Wouldn't this make it harder to distinguish disabled options from enabled ones when you're navigating?

Reply:
Hmm, that's a good point. This PR only changes 1 thing about the focus state for disabled options. I make it so the opacity change doesn't apply to the background of disabled options. That way the focus state can be visible. It makes disabled and enabled options look the same for focus states. 

Here's some screenshots to clarify.

## Appearance A
As implemented in this PR, where disabled has same focus state as enabled.

_coke_ focused
<img width="387" alt="image" src="https://user-images.githubusercontent.com/7720245/224887163-29a6352f-9550-4792-a4c6-1576cc0a582f.png">

_water_ focused
<img width="305" alt="image" src="https://user-images.githubusercontent.com/7720245/224887195-67b1cf93-49f2-4ed7-bbbd-433bdec7701e.png">

## B
Without this code change to the opacity. This causes the `$content-disabled-opacity` to apply to the background of the option, which is used to make the focus state.

_coke_ focused. It's really hard to tell, but _coke_ does have the background change applied, but at a really low opacity.
<img width="328" alt="image" src="https://user-images.githubusercontent.com/7720245/224887003-2ff5532d-f75a-4c82-91e8-71a2be764ed4.png">

_water_ focused
<img width="366" alt="image" src="https://user-images.githubusercontent.com/7720245/224887047-32843f0f-3bea-4503-bbab-542cab230528.png">


AFAIK the focus state of disabled options was not specified before. What do you think would look more appropriate for this PR?

Reply:
Looks like this is resolved, so I'll mark it as resolved. I updated the code comment to try to make it more clear what this SASS is doing, and *why* it does it that way. Feel free to reopen or post a new comment is something doesn't look right.